### PR TITLE
Rubocop rules related to indentation and alignment

### PR DESCRIPTION
**Pull request by exbinary**
### Why
Building on #935, enables rules related to indentation and alignment, fixing corresponding violations.

### What
See the commit list for rules tackled in this PR. The commit comments include links to the corresponding rubocop docs.

Some rules worth highlighting:
* Disables one StandardRB rule: [`Layout/HashAlignment`](https://docs.rubocop.org/rubocop/cops_layout.html#layouthashalignment). See commit comments for my rationale and discussion c00c9bb.
* StandardRB configures [`ArgumentAlignment`](https://docs.rubocop.org/rubocop/cops_layout.html#layoutargumentalignment) in the `with_fixed_indentation` style, which means any multi-line args will be indented relative to the first line, _not_ aligned with the first argument. I found this worked well in our codebase but curious if anyone has differring opinions. See 1cae87d for examples.
* Similarly ^^ for [`ArrayAlignment`](https://docs.rubocop.org/rubocop/cops_layout.html#layoutarrayalignment).
* I find [`FirstArgumentIndentation`](https://docs.rubocop.org/rubocop/cops_layout.html#layoutfirstargumentindentation) in the `consistent` style recommended by StandardRB to be a bit stifling, but not enough to override.
* In the StandardRB style for [`MultilineMethodCallIndentation`](https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilinemethodcallindentation), multi-line method chains are indented two spaces relative to line of the first receiver. This is different from a common style in our codebase where lines are aligned on the 'dots'. Both styles have tradeoffs so i've stuck with StandardRB on this. See ebfa676 for examples.
* StandardRB recommends the `symmetrical` style for [MultilineMethodCallBraceLayout](https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilinemethodcallbracelayout). There are a few places where it's nice to be able to keep one or two args on the same line as the method call but put the rest on separate lines. Good examples are `FactoryBot.create` calls and AR `has_many` callls. In these cases, i've been able to get around this rule by using explicit braces around the trailing hash arg. See 607ec35 for examples.

Note: here's the [StandardRB rubocop config](https://github.com/testdouble/standard/blob/master/config/base.yml).

### How
Please review each commit separately and see commit comments for any related explanation/detail.

### Testing
Relying on our existing suite to catch any regressions.

### Next Steps
?

### Outstanding Questions, Concerns and Other Notes
Open to thoughts about any of the rules enabled here!

### Pre-Merge Checklist
- [x] Security & accessibility have been considered
- [x] Tests have been added, or an explanation has been given why the features cannot be tested
- [x] Documentation and comments have been added to the codebase where required
- [ ] Entry added to CHANGELOG.md if appropriate
- [ ] Outstanding questions and concerns have been resolved
- [ ] Any next steps have been turned into Issues or Discussions as appropriate

### CODE REVIEW

File: app/controllers/matches_controller.rb [hunk 1]
```diff
@@ -61,11 +59,12 @@ def update
     update_connections = params[:commit]&.downcase&.include?('edit who this match connects')
     if @match.update(match_params)
       if save_and_continue || update_connections
-        redirect_to edit_match_path(@match, edit_connection_mode: update_connections ? true : false),
-                    notice: 'Match was successfully updated.'
+        redirect_to(
+          edit_match_path(@match, edit_connection_mode: update_connections ? true : false),
```
Comment (h-m-m) [inline app/controllers/matches_controller.rb:63]: @exbinary — I didn't think any of the rules you mentioned in the discription would add the `(` after `redirect_to` here. Do you know which one does this?

Comment (exbinary) [reply app/controllers/matches_controller.rb:63]: Ah yes, good question! This is something i tweaked manually. The rule at work here is `Layout/ArgumentAlignment`, which forces any multi-line arguments to be indented one level after the method call, not aligned under the first argument as in the original code. But with that automated change, i find the code to be less scannable:
```ruby
# original form -- disallowed by Layout/ArgumentAlignment
redirect_to edit_match_path(@match, edit_connection_mode: update_connections ? true : false),
            notice: 'Match was successfully updated.'

# autocorrected form -- i find this to be imbalanced and harder to scan
redirect_to edit_match_path(@match, edit_connection_mode: update_connections ? true : false),
  notice: 'Match was successfully updated.'

# balanced form -- i prefer this form where the first argument is on a newline, but it requires parentheses
redirect_to(
  edit_match_path(@match, edit_connection_mode: update_connections ? true : false),
  notice: 'Match was successfully updated.'
)
```
Does that make sense? Curious your thoughts on this.

Comment (h-m-m) [reply app/controllers/matches_controller.rb:63]: In discussing this further with @exbinary, I realized my objection was that I keep misreading this as a three-argument call to `edit_match_path` that is indented weirdly. I have no objection to the guideline that's used here, and my confusion is more due to the ternary operator or the fact that `redirect_to` is multi-line when its nested `edit_match_path` isn't

**Comment by exbinary** [comment]
Reviewed in person with @h-m-m and @svileshina .

**Pull request merged**